### PR TITLE
Add lookup and update without changing order in LRU cache

### PR DESCRIPTION
**Pull request by sethiay**
### Description
- Add lookup and update without changing order in LRU cache
- Also removed one of the unit test in job_test.go as it is expected that GetCacheHandle will insert fileInfo in fileInfoCache at the time of creating cache handle and initializing job.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - Added unit test.
3. Integration tests - NA

### CODE REVIEW

File: internal/cache/file/downloader/job.go [hunk 1]
```diff
@@ -175,7 +176,7 @@ func (job *Job) notifySubscribers() {
 	for subItr := job.subscribers.Front(); subItr != nil; subItr = nextSubItr {
 		subItrValue := subItr.Value.(jobSubscriber)
 		nextSubItr = subItr.Next()
-		if job.status.Name == FAILED || job.status.Name == CANCELLED || job.status.Offset >= subItrValue.subscribedOffset {
+		if job.status.Name == FAILED || job.status.Name == CANCELLED || job.status.Name == INVALID || job.status.Offset >= subItrValue.subscribedOffset {
```
Comment (raj-prince) [inline internal/cache/file/downloader/job.go:179]: We can change the `Name` to `State`.

File: internal/cache/lru/lru.go [hunk 2]
```diff
@@ -214,3 +218,49 @@ func (c *Cache) LookUp(key string) (value ValueType) {
 	// Return the value.
 	return e.Value.(entry).Value
 }
+
+// LookUpWithoutChangingOrder looks up previously-inserted value for a given key
+// without changing the order of entries in the cache. Return nil if no value
+// is present.
+func (c *Cache) LookUpWithoutChangingOrder(key string) (value ValueType) {
```
Comment (raj-prince) [inline internal/cache/lru/lru.go:225]: Should we use `sync.RWMutex` to allow multiple readers together for this method?

**Review by raj-prince — CHANGES_REQUESTED** [review]
Looks good. Added a couple of comments.

File: internal/cache/lru/lru.go (hunk 2 above)
Comment (sethiay) [reply internal/cache/lru/lru.go:227]: That's a good catch. Thanks !

File: internal/locker/rw_locker.go [hunk 3]
```diff
@@ -78,7 +78,9 @@ func (c *rwChecker) RUnlock() {
 	c.locker.RUnlock()
 }
 
-// Note: rwDebugger doesn't check potential deadlock in case of read only lock.
+// Note: rwDebugger doesn't check potential deadlock in case of read only lock as
+// doing that is not straight forward because multiple readers can acquire locks
+// at the time same time and that needs keeping track of every different lock.
```
Comment (raj-prince) [inline internal/locker/rw_locker.go:83]: nit: time same time -> same time

**Review by raj-prince — APPROVED** [review]

**Pull request merged**